web/src/app/components/job-list-item/job-list-item.component.ts

```diff
@@ -201,7 +201,12 @@ export class JobListItemComponent implements OnInit, OnDestroy {
   }
 
   selectLoi(node: DynamicFlatNode) {
-    this.navigationService.selectLocationOfInterest(node.loi!.id);
+    if (node.loi) {
```

Comment:
If I understand correctly if `node` has `loi`, that means it's a `loi node` and thus we select LOI, right? If so, can we do like `if (isLoiNode(node))`? It's a very small change, I think it helps the reader understand how DynamicFlatNode works.

Same thing for below.